test/MarketMaker.js

```diff
@@ -295,44 +303,57 @@ contract('marketMaker', function(accounts) {
     const startingSellPrice = 2100;
     const sellPriceIncrement = 1.01;
 
+    token = await simulateCrowdsale(100, [40,30,20,10,0], accounts);
+
     const startBlock = web3.eth.blockNumber + 10;
     const blocksPerPeriod = 100;
 
     var customer = accounts[2];
-    token = await simulateCrowdsale(100, [40,30,20,10,0], accounts);
+    const customerInitEth = web3.eth.getBalance(customer);
+    const customerInitTokens = await token.balanceOf(customer);
+    const priceFactor = 100000;
+
     mm = await LifMarketMaker.new(
       token.address, startBlock, blocksPerPeriod, 24, accounts[1],
-      {value: web3.toWei(8, 'ether'), from: accounts[0]}
+      {value: startingMMBalance, from: accounts[0]}
     );
+    await mm.calculateDistributionPeriods({from: accounts[0]});
 
 
     // Month 0
     await waitForMonth(0, startBlock, blocksPerPeriod);
     // MMETH = 200,   TP = 0,    MMT = 0,   TC = 100, SP = 2100 mETH/Lif, BP = 2000, CL 0%,  maxClaimable = 0,   claimed = 0
-    checkScenarioProperties({
-      marketMakerEthBalance: 200, marketMakerLifBalance: 0, totalProfit: 0,
-      customerEthBalance: 500, customerLifBalance: 100,
-      marketMakerSellPrice: 2100, marketMakerBuyPrice: 2000,
+    await checkScenarioProperties({
+      marketMakerEthBalance: startingMMBalance, marketMakerLifBalance: 0, totalWeiProfit: 0,
+      customerEthBalance: customerInitEth, customerLifBalance: customerInitTokens,
+      marketMakerSellPrice: 2.1*priceFactor, marketMakerBuyPrice: 2*priceFactor,
       claimablePercentage: 0, maxClaimableEth: 0, totalClaimedEth: 0
     }, mm, customer);
 
     // Sell 10 tokens to the MM
-    await token.approve(mm.address, 10);
-    await mm.sendTokens(10, {from: customer});
+    await token.approve(mm.address, help.lif2LifWei(10), {from: customer});
+    await mm.sendTokens(help.lif2LifWei(10), {from: customer});
+
+    help.debug("customer balance (init/now)", customerInitEth, web3.eth.getBalance(customer));
+
     // MMETH = 180,   TP = 0,    MMT = 10,  TC = 90,  SP = 2100 mETH/Lif, BP = 2000, CL 0%,  maxClaimable = 0,   claimed = 0
-    checkScenarioProperties({
-      marketMakerEthBalance: 180, marketMakerLifBalance: 10, totalProfit: 0,
-      customerEthBalance: 520, customerLifBalance: 90,
-      marketMakerSellPrice: 2100, marketMakerBuyPrice: 2000,
+    await checkScenarioProperties({
+      marketMakerEthBalance: web3.toWei(180, 'ether'),
+      marketMakerLifBalance: help.lif2LifWei(10),
+      totalWeiProfit: 0,
+      customerEthBalance: customerInitEth.plus(web3.toWei(20)), customerLifBalance: 90,
+      marketMakerSellPrice: 2.1 * priceFactor, marketMakerBuyPrice: 2 * priceFactor,
```

Comment:
maybe if we do all the proper transformations in the checkScenarioProperties method, it will be bit more readable?